mantidproject/mantid
# Raise exception if no test code in doctest

Added additional information if doctests are run but no test code has been found in the specified doc test. This exception should only be raised if **ALL** given doc tests do not contain test code. 

**To test:**

* Run the doc tests for an algorithm that contains a test.
* Remove the test code and rerun the algorithm (The `.. testcode::` part)
* Ensure you see the new error message included in the output:
   * ``` 
         ...
         Exception occurred:
         File "/home/osboxes/git/mantid/docs/sphinxext/mantiddoc/doctest.py", line 294, in __parse
         raise RuntimeError(NO_TEST_ERROR)
         RuntimeError: 
         *************************************************
         * No test code has been found in given file(s). * 
         *************************************************
        ...
     ```
* Try doing this with a combination of tests:
 * a set of tests where all tests have code - All should run fine
 * a set of tests where some have code and some don't - Only some of the tests should run (no error will be produced)
 * a set where no of the docs contain tests - Should produce same error as one test without test code. 

Fixes #20839

**Release Notes** 
*Internal change - not required in release notes*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
